torch_geometric/data/batch.py

```diff
@@ -210,3 +305,98 @@ def __len__(self) -> int:
     def __reduce__(self):
         state = self.__dict__.copy()
         return DynamicInheritanceGetter(), self.__class__.__bases__, state
+
+    def add_node_attr(self, attrname: str, attr: Tensor) -> None:
```

Comment:
Can we use `setattr` functionality for this?

Reply:
So the issues is, that we do not know if the attribute is a graph, node or edge attribute and thus cannot construct the `_slice_dict` and `_inc_dict`.
I could merge the `add_node_attr` / `add_graph_attr` / `set_edge_attr` into one function with switch if that helps?

Reply:
I merged `add_node_attr` / `add_graph_attr` / `set_edge_attr` into `set_attr`.